app/src/main/java/com/eveningoutpost/dexdrip/EditAlertActivity.java

```diff
@@ -350,30 +357,52 @@ private boolean verifyThreshold(double threshold) {
         return true;
     }
 
+    private double parseDouble(String str) {
+        try {
+            final DecimalFormat numberFormatter = getNumberFormatter(doMgdl);
+            return numberFormatter.parse(str).doubleValue();
+        } catch (NumberFormatException nfe) {
+            Log.e(TAG, "Invalid number", nfe);
+            Toast.makeText(getApplicationContext(), "Invalid number: " + str, Toast.LENGTH_LONG).show();
+            return Double.NaN;
+        } catch (ParseException e) {
+            Log.e(TAG, "Invalid number", e);
+            Toast.makeText(getApplicationContext(), "Invalid number: " + str, Toast.LENGTH_LONG).show();
+            return Double.NaN;
+        }
+    }
+
+    private Integer parseInt(String str) {
+        try {
+            return Integer.parseInt(str);
+        }
+        catch (NumberFormatException nfe) {
+            Log.e(TAG, "Invalid number", nfe);
+            Toast.makeText(getApplicationContext(), "Invalid number: " + str, Toast.LENGTH_LONG).show();
+            return null;
+        }
+    }
+
     public void addListenerOnButtons() {
 
         buttonSave.setOnClickListener(new View.OnClickListener() {
             public void onClick(View v) {
-
                 // Check that values are ok.
-                double threshold = 0;
-                try {
-                    threshold = Double.parseDouble((alertThreshold.getText().toString()));
-                }
-                catch (NumberFormatException nfe) {
-                    Log.e(TAG, "Invalid number", nfe);
-                }
+                double threshold = parseDouble(alertThreshold.getText().toString());
+                if(Double.isNaN(threshold))
+                    return;
+
                 threshold = unitsConvertFromDisp(threshold);
                 if(!verifyThreshold(threshold)) {
                     return;
                 }
+
                 alertReraise = 1;
-                try {
-                    alertReraise = Integer.parseInt((reraise.getText().toString()));
-                }
-                catch (NumberFormatException nfe) {
-                    Log.e(TAG, "Invalid number", nfe);
-                }
+                Integer alterReraiseInt = parseInt(reraise.getText().toString());
+                if(alterReraiseInt ==null)
+                    return;
+                alertReraise = alterReraiseInt;
```

Comment:
any reason for the alterReraiseInt existing instead of just setting alertReraise off the bat?